velox/expression/SignatureBinder.cpp

```diff
@@ -127,15 +127,32 @@ bool SignatureBinder::tryBind(
   if (it == bindings_.end()) {
     // concrete type
     auto typeName = boost::algorithm::to_upper_copy(typeSignature.baseType());
-    if (isDecimalName(typeName)) {
-      VELOX_USER_FAIL("Use 'DECIMAL' in the signature.");
+
+    if (isDecimalName(typeName) && typeName != actualType->kindName()) {
```

Comment:
It would be nice to add test cases for the new functionality to SingatureBinderTest.cpp